src/RokuDeploy.ts

```diff
@@ -652,15 +653,54 @@ export class RokuDeploy {
             throw new errors.UnparsableDeviceResponseError('Invalid response', results);
         }
 
+        this.verboseLog(results.body);
+
         if (results.response.statusCode === 401) {
             throw new errors.UnauthorizedDeviceResponseError('Unauthorized. Please verify username and password for target Roku.', results);
         }
 
+        let rokuMessages = this.getRokuMessagesFromResponseBody(results.body);
+
+        if (rokuMessages.errors.length > 0) {
+            throw new errors.FailedDeviceResponseError(rokuMessages.errors[0], rokuMessages);
+        }
+
         if (results.response.statusCode !== 200) {
-            throw new errors.InvalidDeviceResponseCodeError('Invalid response code: ' + results.response.statusCode);
+            throw new errors.InvalidDeviceResponseCodeError('Invalid response code: ' + results.response.statusCode, results);
         }
     }
 
+    public getRokuMessagesFromResponseBody(body: string): { errors: Array<string>; infos: Array<string>; successes: Array<string> } {
+        let errors = [];
+        let infos = [];
+        let successes = [];
+        let errorRegex = /Shell\.create\('Roku\.Message'\)\.trigger\('[\w\s]+',\s+'(\w+)'\)\.trigger\('[\w\s]+',\s+'(.*?)'\)/igm;
```

Comment:
I know this is how they are sending it currently, but what are your thoughts on matching optional whitespace around all the javascript that could change. such as:
```typescript
/Shell\s*\.\s*create\s*\(\s*'Roku\.Message'\s*\)   ......etc/
```

Reply:
I think this is very very unlinkely. If they do make a change that breaks this it's likely not going to be white space. It's going to be function names and such.